py/dml/structure.py

```diff
@@ -2199,6 +2200,47 @@ def explode_offsets(reg):
             if a[1] > b[0]:
                 report(EREGOL(a[2], b[2], a[3], b[3]))
 
+def check_register_fields(reg):
+    assert dml.globals.dml_version != (1, 2)
+    bitsize = param_int(reg, 'bitsize')
+    def explode_ranges(field):
+        param = field.get_component('lsb')
+        assert param.objtype == 'parameter'
+        ranges = []
+        for field_indices in itertools.product(
+                *([mkIntegerLiteral(field.site, i) for i in range(dimsize)]
+                  for dimsize in field.dimsizes[reg.dimensions:])):
+            reg_indices = tuple(StaticIndex(field.site, var)
+                                for var in reg.idxvars())
+            indices = reg_indices + field_indices
+            lsb_expr = param_expr(field, 'lsb', indices)
+            msb_expr = param_expr(field, 'msb', indices)
+            for expr in (lsb_expr, msb_expr):
+                if isinstance(expr, NonValue):
+                    raise expr.exc()
+                if expr.constant and not isinstance(expr.value, int):
+                    raise EBTYPE(expr.site, expr.ctype(), typestr)
+
+            if lsb_expr.constant and msb_expr.constant:
+                lsb = lsb_expr.value
+                msb = msb_expr.value
+                if lsb < 0 or bitsize <= msb:
+                    raise EBITRR(field)
+                if msb < lsb:
+                    raise EBITRN(field, msb, lsb)
+                ranges.append((lsb, msb, indices))
+        return ranges
+
+    # list of quadruples (lsb, msb, DMLObject, indices)
+    all_ranges = []
+    for field in reg.get_recursive_components('field'):
+        all_ranges.extend((lsb, msb, field, indices)
+                          for (lsb, msb, indices) in explode_ranges(field))
+    all_ranges.sort(key=lambda t: t[0])
```

Comment:
you could consider `yield` and/or `sorted((lsb, ...) for field in .. for (..) in explode(..))`.

Reply:
Not sure. I'll hum-haw over it. Any of the three choices would be fine in my eyes.

Reply:
I have decided I don't care unless you insist.

Reply:
All choices are by far over the threshold for acceptable. I insist that the other choices are slightly better, but I don't insist that you care :)